fuel-tx/src/tests/valid_cases/transaction.rs

```diff
@@ -136,6 +116,104 @@ fn maturity() {
     assert_eq!(CheckError::TransactionMaturity, err);
 }
 
+#[test]
+fn witness_limit() {
```

Comment:
Could we give this name more context? Like, its testing an expected failure mode, it would help of the test name communicated why this is failing.
e.g.
`check__fails_when_too_many_witnesses` or whatever. 

"Given, when, then" would also help. That might mean these should be separate tests.